pkg/reconciler/instances/istio/action.go

```diff
@@ -201,6 +201,16 @@ func canUninstall(istioVersion actions.IstioVersion) bool {
 	return isInstalled(istioVersion) && istioVersion.ClientVersion != ""
 }
 
+func (h *helperVersion) isDowngradePermitted(second *helperVersion) bool {
```

Comment:
I don't like the fact you bind some `downgrade` scenario to the `helperVersion` struct. How about such approach:
- for `helperVersion` you can define a find `differsAtMostByOneMinor()` that will check exactly what the function says,
- for checking the permissible downgrade you can introduce just a local private function or the one that will be attached to the action and inside you will just call the function defined in the previous step, such as:

```go
func (a *ReconcileAction) isDowngradePermitted(current, target helperVersion) bool {
  // implementation details...
}
```

EDIT: I just realised we do already have such function in place and this one is called: `maxOneMinorBehind()` so please reuse it.

Reply:
Will do!